zerver/data_import/rocketchat.py

```diff
@@ -0,0 +1,673 @@
+        user_dict["is_mirror_dummy"] = False
+
+        # RocketChat has three user types:
+        # "user": This is a regular user of the system.
+        # "bot": A special user types for bots.
+        # "unknown": This usually represents a livechat guest.
+        if user_dict["type"] != "user":
+            user_dict["is_mirror_dummy"] = True
+            emails = user_dict.get("emails")
+            if emails is None or len(emails) == 0:
+                user_dict["emails"] = [
+                    {
+                        "address": "{}-{}@{}".format(
+                            user_dict["username"], user_dict["type"], domain_name
+                        )
+                    }
+                ]
+
+        # TODO: Change this to use actual exported avatar
+        avatar_source = "G"
+        full_name = user_dict["name"]
+        id = user_id_mapper.get(rc_user_id)
+        delivery_email = user_dict["emails"][0]["address"]
+        email = user_dict["emails"][0]["address"]
+        short_name = user_dict["username"]
+        date_joined = int(timezone_now().timestamp())
+        timezone = "UTC"
+
+        role = UserProfile.ROLE_MEMBER
+        if "admin" in user_dict["roles"]:
+            role = UserProfile.ROLE_REALM_OWNER
+        elif "guest" in user_dict["roles"]:
+            role = UserProfile.ROLE_GUEST
+
+        if user_dict["is_mirror_dummy"]:
+            is_active = False
+            is_mirror_dummy = True
+        else:
+            is_active = True
+            is_mirror_dummy = False
+
+        user = build_user_profile(
+            avatar_source=avatar_source,
+            date_joined=date_joined,
+            delivery_email=delivery_email,
+            email=email,
+            full_name=full_name,
+            id=id,
+            is_active=is_active,
+            role=role,
+            is_mirror_dummy=is_mirror_dummy,
+            realm_id=realm_id,
+            short_name=short_name,
+            timezone=timezone,
+        )
+        user_handler.add_user(user)
+
+
+def convert_channel_data(
+    room_id_to_room_map: Dict[str, Dict[str, Any]],
+    team_id_to_team_map: Dict[str, Dict[str, Any]],
+    stream_id_mapper: IdMapper,
+    realm_id: int,
+) -> List[ZerverFieldsT]:
+    streams = []
+
+    for rc_room_id in room_id_to_room_map:
+        channel_dict = room_id_to_room_map[rc_room_id]
+
+        now = int(timezone_now().timestamp())
+        stream_id = stream_id_mapper.get(rc_room_id)
+        invite_only = channel_dict["t"] == "p"
+
+        stream_name = channel_dict["name"]
+        stream_desc = channel_dict.get("description", "")
+        if channel_dict.get("teamId"):
+            if channel_dict.get("teamMain") is True:
+                stream_name = "[TEAM] " + stream_name
+            else:
+                stream_desc = "[Team {} channel]. {}".format(
+                    team_id_to_team_map[channel_dict["teamId"]]["name"], stream_desc
+                )
```

Comment:
What's the thinking for the `[TEAM] {stream_name}` formatting of the team channel name?  I think it'd be reasonable to not change stream names and just do something with the stream descriptions.  But maybe I'm missing something about how "teams" work in Rocket.Chat and the stream names aren't themselves globally unique?

Reply:
If we need to combine the channel and team names for uniqueness, I think what I'd want is a format for stream names like `#{channel_name} ({team})`

Reply:
Actually, teams have a separate category in Rocket.Chat, where all the main channels of teams are grouped together. (All teams have one main channel that appears in that category and then several child channels that appears in the normal `Public` and `Private` category.)

![Screenshot from 2021-07-20 03-13-53](https://user-images.githubusercontent.com/39924567/126231428-5782dc2a-9e25-4f06-a02f-1cbb23f0c8ce.png)


So, in order to give the users migrating from Rocket.Chat a similar experience, I added `[TEAM]` before the stream name of the main team channels, which will group all the teams together at the top of the streams list, just like in Rocket.Chat.